<a href="https://colab.research.google.com/github/angel870326/Monthly-Revenue-Forecasting/blob/main/021_announcement_date_v1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

> 2023.06.02 Ssu-Yun Wang<br/>
[Github @angel870326](https://github.com/angel870326)

# **Announcement Date**

### Contents

*   Raw Data
*   Reformat
*   Industries
    *   TSE 上市櫃公司產業別
    *   分為金融業與非金融業

*   Filter Data
    *   Main Data
    *   Select Companies by Main Data
    *   Announcement Date Not Having a Day Value of 10
    *   Announcement Date Not Having a Day Value of 10



In [1]:
# sConnect to the Google Drive
from google.colab import drive
drive.mount("/content/gdrive")

Mounted at /content/gdrive


In [2]:
import os
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [3]:
path = '/content/gdrive/MyDrive/論文/資料集'

## **1. Raw Data**

資料名稱：月營收盈餘

資料期間：1988年1月至2023年1月

資料範圍：上市、櫃公司

資料來源：TEJ Company DB

In [4]:
raw_data = pd.read_excel(os.path.join(path, 'raw/198801-202301 上市櫃公司月營收.xlsx'))
del raw_data['以合併為主單月營收(千元)']
raw_data = raw_data[raw_data["公司"].str.contains('^[a-zA-Z]') == False] # 刪除指數
raw_data

公司       年月       營收發布日
0        1101 台泥  2023/01  2023/02/10
1        1102 亞泥  2023/01  2023/02/10
2        1103 嘉泥  2023/01  2023/02/10
3        1104 環泥  2023/01  2023/02/10
4        1108 幸福  2023/01  2023/02/10
...          ...      ...         ...
458474   2904 匯僑  1988/01         NaN
458475   2905 三商  1988/01         NaN
458476   2913 農林  1988/01         NaN
458477  2915 潤泰全  1988/01         NaN
458478   9902 台火  1988/01         NaN

[442642 rows x 3 columns]

## **2. Reformat**

In [5]:
# Change "Y/m" into "Y-m"
raw_data['年月'] = raw_data['年月'].str.replace('/','-')
raw_data

公司       年月       營收發布日
0        1101 台泥  2023-01  2023/02/10
1        1102 亞泥  2023-01  2023/02/10
2        1103 嘉泥  2023-01  2023/02/10
3        1104 環泥  2023-01  2023/02/10
4        1108 幸福  2023-01  2023/02/10
...          ...      ...         ...
458474   2904 匯僑  1988-01         NaN
458475   2905 三商  1988-01         NaN
458476   2913 農林  1988-01         NaN
458477  2915 潤泰全  1988-01         NaN
458478   9902 台火  1988-01         NaN

[442642 rows x 3 columns]

In [6]:
# 營收發布日 str to datetime
raw_data['營收發布日'] = pd.to_datetime(raw_data['營收發布日'])
raw_data

公司       年月      營收發布日
0        1101 台泥  2023-01 2023-02-10
1        1102 亞泥  2023-01 2023-02-10
2        1103 嘉泥  2023-01 2023-02-10
3        1104 環泥  2023-01 2023-02-10
4        1108 幸福  2023-01 2023-02-10
...          ...      ...        ...
458474   2904 匯僑  1988-01        NaT
458475   2905 三商  1988-01        NaT
458476   2913 農林  1988-01        NaT
458477  2915 潤泰全  1988-01        NaT
458478   9902 台火  1988-01        NaT

[442642 rows x 3 columns]

In [7]:
# Check if there is NaN in each column (241179)
raw_data_nan = pd.DataFrame(raw_data.isnull().sum(), columns=['nan_count'])
raw_data_nan['nan_percent'] = raw_data_nan['nan_count'] / len(raw_data)
raw_data_nan

nan_count  nan_percent
公司             0     0.000000
年月             0     0.000000
營收發布日     241179     0.544862

In [8]:
# Pivot table
raw_data_pvt = raw_data.pivot_table(values='營收發布日', 
		                                index='公司', 
	   	                              columns='年月')
del raw_data_pvt['2023-01']  # delete 2023-01
raw_data_pvt

年月          2013-01    2013-02    2013-03    2013-04    2013-05    2013-06  \
公司                                                                           
1101 台泥  2013-02-18 2013-03-08 2013-04-10 2013-05-10 2013-06-10 2013-07-10   
1102 亞泥  2013-02-08 2013-03-08 2013-04-10 2013-05-10 2013-06-10 2013-07-10   
1103 嘉泥  2013-02-18 2013-03-11 2013-04-10 2013-05-10 2013-06-10 2013-07-10   
1104 環泥  2013-02-08 2013-03-08 2013-04-10 2013-05-10 2013-06-10 2013-07-10   
1108 幸福  2013-02-18 2013-03-08 2013-04-09 2013-05-10 2013-06-07 2013-07-10   
...             ...        ...        ...        ...        ...        ...   
9951 皇田  2013-02-07 2013-03-07 2013-04-09 2013-05-10 2013-06-07 2013-07-08   
9955 佳龍  2013-02-18 2013-03-08 2013-04-09 2013-05-10 2013-06-10 2013-07-09   
9958 世紀鋼 2013-02-08 2013-03-10 2013-04-09 2013-05-10 2013-06-10 2013-07-10   
9960 邁達康 2013-02-18 2013-03-08 2013-04-10 2013-05-10 2013-06-10 2013-07-10   
9962 有益  2013-02-08 2013-03-08 2013-04-10 2013-05-10 2013-06-07 2013-07-05   

年月          2013-07    2013-08    2013-09    2013-10  ...    2022-03  \
公司                                                    ...              
1101 台泥  2013-08-09 2013-09-10 2013-10-11 2013-11-11  ... 2022-04-11   
1102 亞泥  2013-08-09 2013-09-10 2013-10-09 2013-11-08  ... 2022-04-08   
1103 嘉泥  2013-08-09 2013-09-10 2013-10-11 2013-11-11  ... 2022-04-11   
1104 環泥  2013-08-09 2013-09-10 2013-10-09 2013-11-08  ... 2022-04-11   
1108 幸福  2013-08-09 2013-09-09 2013-10-08 2013-11-08  ... 2022-04-11   
...             ...        ...        ...        ...  ...        ...   
9951 皇田  2013-08-08 2013-09-06 2013-10-08 2013-11-06  ... 2022-04-08   
9955 佳龍  2013-08-09 2013-09-10 2013-10-09 2013-11-08  ... 2022-04-08   
9958 世紀鋼 2013-08-09 2013-09-10 2013-10-09 2013-11-08  ... 2022-04-11   
9960 邁達康 2013-08-10 2013-09-10 2013-10-09 2013-11-08  ... 2022-04-08   
9962 有益  2013-08-09 2013-09-06 2013-10-08 2013-11-07  ... 2022-04-08   

年月          2022-04    2022-05    2022-06    2022-07    2022-08    2022-09  \
公司                                                                           
1101 台泥  2022-05-10 2022-06-10 2022-07-11 2022-08-10 2022-09-12 2022-10-06   
1102 亞泥  2022-05-10 2022-06-10 2022-07-08 2022-08-10 2022-09-08 2022-10-07   
1103 嘉泥  2022-05-10 2022-06-10 2022-07-11 2022-08-10 2022-09-12 2022-10-11   
1104 環泥  2022-05-10 2022-06-10 2022-07-08 2022-08-10 2022-09-12 2022-10-11   
1108 幸福  2022-05-09 2022-06-09 2022-07-08 2022-08-08 2022-09-08 2022-10-07   
...             ...        ...        ...        ...        ...        ...   
9951 皇田  2022-05-10 2022-06-09 2022-07-08 2022-08-09 2022-09-12 2022-10-11   
9955 佳龍  2022-05-10 2022-06-10 2022-07-06 2022-08-10 2022-09-06 2022-10-07   
9958 世紀鋼 2022-05-09 2022-06-09 2022-07-11 2022-08-10 2022-09-08 2022-10-07   
9960 邁達康 2022-05-10 2022-06-10 2022-07-08 2022-08-10 2022-09-09 2022-10-07   
9962 有益  2022-05-09 2022-06-08 2022-07-08 2022-08-08 2022-09-07 2022-10-06   

年月          2022-10    2022-11    2022-12  
公司                                         
1101 台泥  2022-11-10 2022-12-09 2023-01-10  
1102 亞泥  2022-11-10 2022-12-09 2023-01-10  
1103 嘉泥  2022-11-10 2022-12-12 2023-01-10  
1104 環泥  2022-11-10 2022-12-09 2023-01-10  
1108 幸福  2022-11-08 2022-12-09 2023-01-10  
...             ...        ...        ...  
9951 皇田  2022-11-10 2022-12-09 2023-01-10  
9955 佳龍  2022-11-09 2022-12-09 2023-01-10  
9958 世紀鋼 2022-11-10 2022-12-09 2023-01-09  
9960 邁達康 2022-11-10 2022-12-09 2023-01-10  
9962 有益  2022-11-04 2022-12-07 2023-01-09  

[1780 rows x 120 columns]

## **3. Filter Data**

### **3.1 Main Data**

【**月營收盈餘 (2013-2022)**】

資料期間：2013年1月至2022年12月（共120個月）

資料範圍：上市、櫃公司（排除金融業、生技醫療、建材營造、DR和KY公司）

資料來源：TEJ Company DB、公開資訊觀測站

In [9]:
org_data = pd.read_excel(os.path.join(path, '007_v1/201301-202212上市櫃公司月營收_非金融業.xlsx'), index_col=0)
org_data.columns = pd.to_datetime(org_data.columns, format="%Y-%m-%d").to_period('M')
org_data

2013-01  2013-02  2013-03  2013-04  2013-05  2013-06  2013-07  \
公司                                                                        
1101 台泥   9134465  5540346  9457971  9919269  9543782  9517630  9875888   
1102 亞泥   6018213  2552357  5428755  5930748  6239676  5952754  5942364   
1103 嘉泥    288455   166638   286007   365292   382601   302995   294781   
1104 環泥    486481   299860   461732   394631   406677   415968   453397   
1108 幸福    481802   276936   444917   362054   381384   368109   439572   
...           ...      ...      ...      ...      ...      ...      ...   
9951 皇田    201785   167967   240746   243935   238296   193880   198427   
9955 佳龍    394489   383183   428478   564053   336622   295391   434605   
9958 世紀鋼   198944   166364   351222   280864   289332   426371   213281   
9960 邁達康    52534    41935    61642    70998    81508    64525    62085   
9962 有益    298876   327002   313861   241566   251782   273664   174682   

          2013-08   2013-09   2013-10  ...  2022-03  2022-04  2022-05  \
公司                                     ...                              
1101 台泥   9835143  10060975  10654077  ...  9971650  8319342  7733787   
1102 亞泥   5786107   5879394   6478670  ...  8160414  8710220  8000427   
1103 嘉泥    336088    314563    429783  ...   220463   168089   163521   
1104 環泥    393203    448691    521445  ...   591593   638493   537082   
1108 幸福    379115    387362    450770  ...   345612   335518   332258   
...           ...       ...       ...  ...      ...      ...      ...   
9951 皇田    256724    228796    250756  ...   374229   302262   323433   
9955 佳龍    306534    266617    363766  ...    96200   101850    95096   
9958 世紀鋼   302589    401695    255738  ...   626104   401960   673479   
9960 邁達康    60960     60309     61582  ...    60275    86754    69752   
9962 有益    238490    165340    202491  ...   366892   248958   228412   

          2022-06   2022-07   2022-08   2022-09   2022-10  2022-11   2022-12  
公司                                                                            
1101 台泥   9145989  10102468  10689860  10404901  11368096  9674576  12584154  
1102 亞泥   7776413   7864622   7069221   6994078   7601097  8306062   8340507  
1103 嘉泥    183177    178825    182371    205264    209429   221763    228644  
1104 環泥    573028    580420    605512    597159    634981   631827    725055  
1108 幸福    334113    326691    390053    346635    401202   383773    418326  
...           ...       ...       ...       ...       ...      ...       ...  
9951 皇田    371791    337581    468608    464373    432835   500111    506796  
9955 佳龍     80726     85625     81881     79179     80630    91270     84115  
9958 世紀鋼   665459    651699    757968    903198    911834   944060   1082675  
9960 邁達康   103280     64983    105969    113755     78996    96570     58764  
9962 有益    387082    248327    318092    264270    307426   308663    385673  

[1240 rows x 120 columns]

In [10]:
print("Data shape:", org_data.shape)
print("Data size:", org_data.size)

Data shape: (1240, 120)
Data size: 148800


### **3.2 Select Companies by Main Data**



In [11]:
filtered_data = raw_data_pvt.loc[org_data.index]
filtered_data

年月          2013-01    2013-02    2013-03    2013-04    2013-05    2013-06  \
公司                                                                           
1101 台泥  2013-02-18 2013-03-08 2013-04-10 2013-05-10 2013-06-10 2013-07-10   
1102 亞泥  2013-02-08 2013-03-08 2013-04-10 2013-05-10 2013-06-10 2013-07-10   
1103 嘉泥  2013-02-18 2013-03-11 2013-04-10 2013-05-10 2013-06-10 2013-07-10   
1104 環泥  2013-02-08 2013-03-08 2013-04-10 2013-05-10 2013-06-10 2013-07-10   
1108 幸福  2013-02-18 2013-03-08 2013-04-09 2013-05-10 2013-06-07 2013-07-10   
...             ...        ...        ...        ...        ...        ...   
9951 皇田  2013-02-07 2013-03-07 2013-04-09 2013-05-10 2013-06-07 2013-07-08   
9955 佳龍  2013-02-18 2013-03-08 2013-04-09 2013-05-10 2013-06-10 2013-07-09   
9958 世紀鋼 2013-02-08 2013-03-10 2013-04-09 2013-05-10 2013-06-10 2013-07-10   
9960 邁達康 2013-02-18 2013-03-08 2013-04-10 2013-05-10 2013-06-10 2013-07-10   
9962 有益  2013-02-08 2013-03-08 2013-04-10 2013-05-10 2013-06-07 2013-07-05   

年月          2013-07    2013-08    2013-09    2013-10  ...    2022-03  \
公司                                                    ...              
1101 台泥  2013-08-09 2013-09-10 2013-10-11 2013-11-11  ... 2022-04-11   
1102 亞泥  2013-08-09 2013-09-10 2013-10-09 2013-11-08  ... 2022-04-08   
1103 嘉泥  2013-08-09 2013-09-10 2013-10-11 2013-11-11  ... 2022-04-11   
1104 環泥  2013-08-09 2013-09-10 2013-10-09 2013-11-08  ... 2022-04-11   
1108 幸福  2013-08-09 2013-09-09 2013-10-08 2013-11-08  ... 2022-04-11   
...             ...        ...        ...        ...  ...        ...   
9951 皇田  2013-08-08 2013-09-06 2013-10-08 2013-11-06  ... 2022-04-08   
9955 佳龍  2013-08-09 2013-09-10 2013-10-09 2013-11-08  ... 2022-04-08   
9958 世紀鋼 2013-08-09 2013-09-10 2013-10-09 2013-11-08  ... 2022-04-11   
9960 邁達康 2013-08-10 2013-09-10 2013-10-09 2013-11-08  ... 2022-04-08   
9962 有益  2013-08-09 2013-09-06 2013-10-08 2013-11-07  ... 2022-04-08   

年月          2022-04    2022-05    2022-06    2022-07    2022-08    2022-09  \
公司                                                                           
1101 台泥  2022-05-10 2022-06-10 2022-07-11 2022-08-10 2022-09-12 2022-10-06   
1102 亞泥  2022-05-10 2022-06-10 2022-07-08 2022-08-10 2022-09-08 2022-10-07   
1103 嘉泥  2022-05-10 2022-06-10 2022-07-11 2022-08-10 2022-09-12 2022-10-11   
1104 環泥  2022-05-10 2022-06-10 2022-07-08 2022-08-10 2022-09-12 2022-10-11   
1108 幸福  2022-05-09 2022-06-09 2022-07-08 2022-08-08 2022-09-08 2022-10-07   
...             ...        ...        ...        ...        ...        ...   
9951 皇田  2022-05-10 2022-06-09 2022-07-08 2022-08-09 2022-09-12 2022-10-11   
9955 佳龍  2022-05-10 2022-06-10 2022-07-06 2022-08-10 2022-09-06 2022-10-07   
9958 世紀鋼 2022-05-09 2022-06-09 2022-07-11 2022-08-10 2022-09-08 2022-10-07   
9960 邁達康 2022-05-10 2022-06-10 2022-07-08 2022-08-10 2022-09-09 2022-10-07   
9962 有益  2022-05-09 2022-06-08 2022-07-08 2022-08-08 2022-09-07 2022-10-06   

年月          2022-10    2022-11    2022-12  
公司                                         
1101 台泥  2022-11-10 2022-12-09 2023-01-10  
1102 亞泥  2022-11-10 2022-12-09 2023-01-10  
1103 嘉泥  2022-11-10 2022-12-12 2023-01-10  
1104 環泥  2022-11-10 2022-12-09 2023-01-10  
1108 幸福  2022-11-08 2022-12-09 2023-01-10  
...             ...        ...        ...  
9951 皇田  2022-11-10 2022-12-09 2023-01-10  
9955 佳龍  2022-11-09 2022-12-09 2023-01-10  
9958 世紀鋼 2022-11-10 2022-12-09 2023-01-09  
9960 邁達康 2022-11-10 2022-12-09 2023-01-10  
9962 有益  2022-11-04 2022-12-07 2023-01-09  

[1240 rows x 120 columns]

### **3.3 Announcement Date Not Having a Day Value of 10**



In [12]:
# Initialize count
count = 0

# Iterate over columns
for column in filtered_data.columns:
    # Iterate over rows
    for value in filtered_data[column]:
        # Check if day is 10
        if value.day != 10:
            count += 1

# Print the count
print(count)
print(count/filtered_data.size)

107145
0.7200604838709678


### **3.4 Check for NaN Values**



In [13]:
# Check if there is NaN in each row
row_nan_count = pd.DataFrame(filtered_data.isna().sum(axis=1), columns=['nan_count'])
nan_row = row_nan_count[row_nan_count['nan_count'] > 0]
nan_row 

nan_count
公司                 
1264 德麥          10
2736 富野           9
3437 榮創           4
4438 廣越          29
4549 桓達           8
4550 長佳           9
4551 智伸科         10
4760 勤凱           5
6425 易發           4
6426 統新           6
6432 今展科          8
6435 大中           8
6438 迅得           9
6441 廣錠          10
6442 光聖          10
6443 元晶          10
6449 鈺邦          12
6582 申豐          40
6790 永豐實         91
8450 霹靂           7
8454 富邦媒         11
8473 山林水         23

### **3.5 Replace the NaN Values with the Date of the 10th Day (YYYY-MM-10)**



In [14]:
# from datetime import datetime, timedelta
from datetime import datetime
from dateutil.relativedelta import relativedelta

In [15]:
# Change a column of string type to datetime type
final_data = filtered_data.copy()
columns = [datetime.strptime(date_str, '%Y-%m').date() for date_str in final_data.columns]
print(columns)

[datetime.date(2013, 1, 1), datetime.date(2013, 2, 1), datetime.date(2013, 3, 1), datetime.date(2013, 4, 1), datetime.date(2013, 5, 1), datetime.date(2013, 6, 1), datetime.date(2013, 7, 1), datetime.date(2013, 8, 1), datetime.date(2013, 9, 1), datetime.date(2013, 10, 1), datetime.date(2013, 11, 1), datetime.date(2013, 12, 1), datetime.date(2014, 1, 1), datetime.date(2014, 2, 1), datetime.date(2014, 3, 1), datetime.date(2014, 4, 1), datetime.date(2014, 5, 1), datetime.date(2014, 6, 1), datetime.date(2014, 7, 1), datetime.date(2014, 8, 1), datetime.date(2014, 9, 1), datetime.date(2014, 10, 1), datetime.date(2014, 11, 1), datetime.date(2014, 12, 1), datetime.date(2015, 1, 1), datetime.date(2015, 2, 1), datetime.date(2015, 3, 1), datetime.date(2015, 4, 1), datetime.date(2015, 5, 1), datetime.date(2015, 6, 1), datetime.date(2015, 7, 1), datetime.date(2015, 8, 1), datetime.date(2015, 9, 1), datetime.date(2015, 10, 1), datetime.date(2015, 11, 1), datetime.date(2015, 12, 1), datetime.date(2016

In [16]:
# Get the indexes of NaN values (row, col)
nan_index = np.where(final_data.isna())
nan_index

(array([  27,   27,   27,   27,   27,   27,   27,   27,   27,   27,  414,
         414,  414,  414,  414,  414,  414,  414,  414,  591,  591,  591,
         591,  721,  721,  721,  721,  721,  721,  721,  721,  721,  721,
         721,  721,  721,  721,  721,  721,  721,  721,  721,  721,  721,
         721,  721,  721,  721,  721,  721,  721,  721,  742,  742,  742,
         742,  742,  742,  742,  742,  743,  743,  743,  743,  743,  743,
         743,  743,  743,  744,  744,  744,  744,  744,  744,  744,  744,
         744,  744,  760,  760,  760,  760,  760, 1051, 1051, 1051, 1051,
        1052, 1052, 1052, 1052, 1052, 1052, 1053, 1053, 1053, 1053, 1053,
        1053, 1053, 1053, 1054, 1054, 1054, 1054, 1054, 1054, 1054, 1054,
        1055, 1055, 1055, 1055, 1055, 1055, 1055, 1055, 1055, 1056, 1056,
        1056, 1056, 1056, 1056, 1056, 1056, 1056, 1056, 1057, 1057, 1057,
        1057, 1057, 1057, 1057, 1057, 1057, 1057, 1058, 1058, 1058, 1058,
        1058, 1058, 1058, 1058, 1058, 

In [17]:
# Replace the NaN values with the date of the 10th day (YYYY-MM-10)
for i in range(0, len(nan_index[0]), 1):
    row_index = nan_index[0][i]
    col_index = nan_index[1][i]
    col_date = columns[col_index]
    final_data.iat[row_index, col_index] = pd.to_datetime(col_date + relativedelta(months=1, day=10))

In [18]:
# Check if there is NaN in each row
row_nan_count = pd.DataFrame(final_data.isna().sum(axis=1), columns=['nan_count'])
nan_row = row_nan_count[row_nan_count['nan_count'] > 0]
nan_row 

Empty DataFrame
Columns: [nan_count]
Index: []

### **3.6 Output Data**


In [19]:
final_data.to_excel(os.path.join(path,'201301-202212 announcement_date.xlsx'))
final_data

年月          2013-01    2013-02    2013-03    2013-04    2013-05    2013-06  \
公司                                                                           
1101 台泥  2013-02-18 2013-03-08 2013-04-10 2013-05-10 2013-06-10 2013-07-10   
1102 亞泥  2013-02-08 2013-03-08 2013-04-10 2013-05-10 2013-06-10 2013-07-10   
1103 嘉泥  2013-02-18 2013-03-11 2013-04-10 2013-05-10 2013-06-10 2013-07-10   
1104 環泥  2013-02-08 2013-03-08 2013-04-10 2013-05-10 2013-06-10 2013-07-10   
1108 幸福  2013-02-18 2013-03-08 2013-04-09 2013-05-10 2013-06-07 2013-07-10   
...             ...        ...        ...        ...        ...        ...   
9951 皇田  2013-02-07 2013-03-07 2013-04-09 2013-05-10 2013-06-07 2013-07-08   
9955 佳龍  2013-02-18 2013-03-08 2013-04-09 2013-05-10 2013-06-10 2013-07-09   
9958 世紀鋼 2013-02-08 2013-03-10 2013-04-09 2013-05-10 2013-06-10 2013-07-10   
9960 邁達康 2013-02-18 2013-03-08 2013-04-10 2013-05-10 2013-06-10 2013-07-10   
9962 有益  2013-02-08 2013-03-08 2013-04-10 2013-05-10 2013-06-07 2013-07-05   

年月          2013-07    2013-08    2013-09    2013-10  ...    2022-03  \
公司                                                    ...              
1101 台泥  2013-08-09 2013-09-10 2013-10-11 2013-11-11  ... 2022-04-11   
1102 亞泥  2013-08-09 2013-09-10 2013-10-09 2013-11-08  ... 2022-04-08   
1103 嘉泥  2013-08-09 2013-09-10 2013-10-11 2013-11-11  ... 2022-04-11   
1104 環泥  2013-08-09 2013-09-10 2013-10-09 2013-11-08  ... 2022-04-11   
1108 幸福  2013-08-09 2013-09-09 2013-10-08 2013-11-08  ... 2022-04-11   
...             ...        ...        ...        ...  ...        ...   
9951 皇田  2013-08-08 2013-09-06 2013-10-08 2013-11-06  ... 2022-04-08   
9955 佳龍  2013-08-09 2013-09-10 2013-10-09 2013-11-08  ... 2022-04-08   
9958 世紀鋼 2013-08-09 2013-09-10 2013-10-09 2013-11-08  ... 2022-04-11   
9960 邁達康 2013-08-10 2013-09-10 2013-10-09 2013-11-08  ... 2022-04-08   
9962 有益  2013-08-09 2013-09-06 2013-10-08 2013-11-07  ... 2022-04-08   

年月          2022-04    2022-05    2022-06    2022-07    2022-08    2022-09  \
公司                                                                           
1101 台泥  2022-05-10 2022-06-10 2022-07-11 2022-08-10 2022-09-12 2022-10-06   
1102 亞泥  2022-05-10 2022-06-10 2022-07-08 2022-08-10 2022-09-08 2022-10-07   
1103 嘉泥  2022-05-10 2022-06-10 2022-07-11 2022-08-10 2022-09-12 2022-10-11   
1104 環泥  2022-05-10 2022-06-10 2022-07-08 2022-08-10 2022-09-12 2022-10-11   
1108 幸福  2022-05-09 2022-06-09 2022-07-08 2022-08-08 2022-09-08 2022-10-07   
...             ...        ...        ...        ...        ...        ...   
9951 皇田  2022-05-10 2022-06-09 2022-07-08 2022-08-09 2022-09-12 2022-10-11   
9955 佳龍  2022-05-10 2022-06-10 2022-07-06 2022-08-10 2022-09-06 2022-10-07   
9958 世紀鋼 2022-05-09 2022-06-09 2022-07-11 2022-08-10 2022-09-08 2022-10-07   
9960 邁達康 2022-05-10 2022-06-10 2022-07-08 2022-08-10 2022-09-09 2022-10-07   
9962 有益  2022-05-09 2022-06-08 2022-07-08 2022-08-08 2022-09-07 2022-10-06   

年月          2022-10    2022-11    2022-12  
公司                                         
1101 台泥  2022-11-10 2022-12-09 2023-01-10  
1102 亞泥  2022-11-10 2022-12-09 2023-01-10  
1103 嘉泥  2022-11-10 2022-12-12 2023-01-10  
1104 環泥  2022-11-10 2022-12-09 2023-01-10  
1108 幸福  2022-11-08 2022-12-09 2023-01-10  
...             ...        ...        ...  
9951 皇田  2022-11-10 2022-12-09 2023-01-10  
9955 佳龍  2022-11-09 2022-12-09 2023-01-10  
9958 世紀鋼 2022-11-10 2022-12-09 2023-01-09  
9960 邁達康 2022-11-10 2022-12-09 2023-01-10  
9962 有益  2022-11-04 2022-12-07 2023-01-09  

[1240 rows x 120 columns]